src/lib/y2partitioner/widgets/fstab_options.rb

```diff
@@ -0,0 +1,531 @@
+          end
+
+          Yast::Wizard.ShowHelp(help.join("\n"))
+        end
+
+        nil
+      end
+
+      def contents
+        VBox(
+          Left(MountBy.new(@options)),
+          VSpacing(1),
+          Left(VolumeLabel.new(@options)),
+          VSpacing(1),
+          Left(GeneralOptions.new(@options)),
+          Left(FilesystemsOptions.new(@options)),
+          * ui_term_with_vspace(JournalOptions.new(@options)),
+          * ui_term_with_vspace(AclOptions.new(@options)),
+          Left(ArbitraryOptions.new(@options))
+        )
+      end
+
+    private
+
+      def widgets
+        Yast::CWM.widgets_in_contents([self])
+      end
+    end
+
+    # Input field to set the partition Label
+    class VolumeLabel < CWM::InputField
+      include FstabCommon
+
+      def label
+        _("Volume &Label")
+      end
+
+      def store
+        @options.label = value
+      end
+
+      def init
+        self.value = @options.label
+      end
+    end
+
+    # Group of radio buttons to select the type of identifier to be used for
+    # mouth the specific device (UUID, Label, Path...)
+    class MountBy < CWM::CustomWidget
+      include FstabCommon
+
+      def label
+        _("Mount in /etc/fstab by")
+      end
+
+      def store
+        @options.mount_by = selected_mount_by
+      end
+
+      def init
+        value = @options.mount_by ? @options.mount_by.to_sym : :uuid
+        Yast::UI.ChangeWidget(Id(:mt_group), :Value, value)
+      end
+
+      def contents
+        RadioButtonGroup(
+          Id(:mt_group),
+          VBox(
+            Left(Label(label)),
+            HBox(
+              VBox(
+                Left(RadioButton(Id(:device), _("&Device Name"))),
+                Left(RadioButton(Id(:label), _("Volume &Label"))),
+                Left(RadioButton(Id(:uuid), _("&UUID")))
+              ),
+              Top(
+                VBox(
+                  Left(RadioButton(Id(:id), _("Device &ID"))),
+                  Left(RadioButton(Id(:path), _("Device &Path")))
+                )
+              )
+            )
+          )
+        )
+      end
+
+      def selected_mount_by
+        Y2Storage::Filesystems::MountByType.all.detect do |fs|
+          fs.to_sym == value
+        end
+      end
+
+      def value
+        Yast::UI.QueryWidget(Id(:mt_group), :Value)
+      end
+    end
+
+    # A group of options that are general for many filesystem types.
+    class GeneralOptions < CWM::CustomWidget
+      include FstabCommon
+
+      def contents
+        return Empty() unless widgets.any? { |w| w.supported_by_filesystem? }
+
+        VBox(* widgets.map { |w| to_ui_term(w) }, VSpacing(1))
+      end
+
+      def widgets
+        [
+          ReadOnly.new(@options),
+          Noatime.new(@options),
+          MountUser.new(@options),
+          Noauto.new(@options),
+          Quota.new(@options)
+        ]
+      end
+    end
+
+    # Generic checkbox for fstab options
```

Comment:
> VALUES must be a pair: ["fire", "water"] means "fire" is checked and "water" unchecked

the logic is backwards. the 0-th element should be falsey and the 1-st should be truthy.
(it is complicated enough by some checkboxes having "no" in their label)